src/models/Package.ts

```diff
@@ -23,11 +29,31 @@ const PackageSchema = new Schema<PackageProps>(
     deliveredAt: { type: Date, default: undefined },
     status: { type: String, default: undefined },
     deadlines: { type: Date, required: true },
+    city: { type: String, required: true },
+    coordinates: {
+      lat: Number,
+      lng: Number,
+    },
   },
   {
     timestamps: true,
   }
 );
+PackageSchema.pre<PackageProps>("save", async function () {
+  try {
+    const { address, city } = this;
+    const geocodeResult = await geocodeAddress(address, city);
+    if (geocodeResult) {
+      const { lat, lng } = geocodeResult;
+
+      this.coordinates = { lat, lng };
+    } else {
+      throw new Error("No coordinates found for the provided address");
+    }
```

Comment:
Acá podrías chequear por lo contrario, y hacer el throw directamente. 

Entonces haces el narrow por una parte y por otra el código te queda sin llaves, digamos le sacas un nivel

```suggestion
    if (!geocodeResult) 
          throw new Error("No coordinates found for the provided address");
          
      const { lat, lng } = geocodeResult;

      this.coordinates = { lat, lng };
```